core/event_loop.go

```diff
@@ -23,25 +23,46 @@ import (
 // EventLoopWithInterrupt starts an instance of the agent created with NewAgent().
 // Agent is stopped when <closeChan> is closed, a user interrupt (SIGINT), or a
 // terminate signal (SIGTERM) is received.
-func EventLoopWithInterrupt(agent *Agent, closeChan chan struct{}) error {
-	err := agent.Start()
-	if err != nil {
-		agent.Error("Error loading core: ", err)
-		return err
-	}
+func EventLoopWithInterrupt(agent *Agent, closeCh chan struct{}) error {
+	_, errCh := EventLoop(agent, closeCh)
 
-	sigChan := make(chan os.Signal, 1)
-	signal.Notify(sigChan, os.Interrupt)
-	signal.Notify(sigChan, syscall.SIGTERM)
 	select {
```

Comment:
`select` is unnecessary here when using single case. You could just do: `return <-errCh`.

Reply:
Good catch.  Fixed pushed :)